examples/tutorial.dx

```diff
@@ -0,0 +1,494 @@
+'# Dex Tutorial
+
+
+'Dex is a functional, statically typed language for array processing.
+There are many tools for array processing from high-level libraries
+like NumPy / MATLAB to low-level languages like CUDA. Dex gives you
+many benefit of the safety and simplicity benefits of high-level array
+processing languages, without requiring that you give up low-level
+control.
+
+'## Array Comprehensions
+
+
+' Before getting into the details of the language, let us begin with
+the most useful component of dex, the `for` builder. The best analogy
+for this construct is list comprehensions in Python. For instance, in
+Python we might write a list comprehension like:
+
+' `x = [[1 for j in range(10)] for i in range(5)]`
+
+' In Dex, this construct would be written as, 
+
+x = for i:(Fin 5). for j:(Fin 10). 1
+
+
+' Once we have an variable we can print it `:p` 
+
+:p x
+
+' More interestingly, we can also see its type with `:t`. This type
+signature tells us that `x` is a two-dimensional array, with first
+dimension of size 5 and the second of size 10.
+
+:t x
+
+' Once we have an array we can use it in new comprehensions. For example,
+  if say we want to add `5` to each element of the array. In Python,
+  you might write this as, 
+
+' `y = [[x[i][j] for j in range(10)] for i in range(5)]`
+
+' Dex can do something similar. The main superficial difference is the
+  indexing syntax which uses `.` instead of brackets.
+
+y = for i:(Fin 5). for j:(Fin 10). x.i.j + 5 
+
+:p y
+
+' However, we can make this expression nicer. Because `x` has a known type
+  and `i` and `j` index into that type, Dex can infer the range of the loop.
+  That means that we can safely remove `Fin` statements and get the same result.
```

Comment:
Great transition into explaining Dex's index set type inference! This was also mentioned in ["A comparison of Futhark and Dex"](https://futhark-lang.org/blog/2020-12-28-futhark-and-dex.html#the-good-ones) as a unique feature of Dex.